receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -49,22 +55,67 @@ func newReceiver(params receiver.CreateSettings, config *Config, consumer consum
 		return nil, err
 	}
 
-	return &k8sobjectsreceiver{
-		client:   client,
-		setting:  params,
-		consumer: consumer,
-		objects:  config.Objects,
-		obsrecv:  obsrecv,
-		mu:       sync.Mutex{},
-	}, nil
+	objReceiver := &k8sobjectsreceiver{
+		client:         client,
+		setting:        params,
+		logger:         params.Logger,
+		consumer:       consumer,
+		objects:        config.Objects,
+		obsrecv:        obsrecv,
+		mu:             sync.Mutex{},
+		leaderElection: config.LeaderElection,
+	}
+
+	if config.LeaderElection.Enabled {
+		objReceiver.leaderElectionClient, err = config.getClient()
+		if err != nil {
+			return nil, err
+		}
+
+		// use component "type-name" as resource lock name.
+		objReceiver.leaderElection.LockName = getLeaderElectionLockName(params.ID)
```

Comment:
So the readme above seems to imply that users can set the lock name with the option `lock_name`, but this is always overriding the user provided name and setting the lock name to the component id, right?

Reply:
Thank you for reminding me, I will update to make a supplementary logic, if the user does not set `luckname`, the default is `component id`